### PR TITLE
Tests for Announcement.jsx, deletePostButton.jsx and Sentiments.jsx

### PR DESCRIPTION
**Pull request by MilanBarande**
This is part of the "let's improve the coverage" epic.

✅ Tests for the Announcement component
✅ Added missing flow annotations on Announcement
✅ Quick refactoring of the Announcement component
✅ Tests for the Sentiments component (TODO)
✅ Tests for the DeletePostButton component
✅ Quick refactoring of DeletePostButton component and CSS

### CODE REVIEW

File: assembl/static2/js/app/components/debate/common/announcement.jsx [hunk 1]
```diff
@@ -23,8 +23,29 @@ type SentimentsCounts = {
...
+
 type Posts = {
-  edges: Array<{ node: { sentimentCounts: { [string]: number }, publicationState: string } }>
+  edges: Array<Post>
+};
+
+type AnnouncementContent = {
+  __typename: string,
+  body: string,
+  title: string
```
Comment (amensouissi) [inline assembl/static2/js/app/components/debate/common/announcement.jsx]: only body is required here!

File: assembl/static2/js/app/components/debate/common/announcement.jsx [hunk 2]
```diff
@@ -77,55 +98,50 @@ const dirtySplitHack = (announcementContent) => {
...
+  return (
+    <div className="announcement">
+      <div className="announcement-title">
+        <div className="title-hyphen">&nbsp;</div>
+        <h3 className="announcement-title-text dark-title-1">
+          <Translate value="debate.thread.announcement" />
+        </h3>
+      </div>
+      <Col xs={12} md={8} className="announcement-media col-md-push-4">
+        {mediaContent && <TextAndMedia {...mediaContent} />}
```
Comment (amensouissi) [inline assembl/static2/js/app/components/debate/common/announcement.jsx]: mediaContent?<TextAndMedia {...mediaContent} />: null

File: assembl/static2/js/app/components/debate/common/announcement.spec.jsx [hunk 3]
```diff
@@ -0,0 +1,100 @@
+import React from 'react';
+import renderer from 'react-test-renderer'; // eslint-disable-line
+
+import Announcement from '../../../../../js/app/components/debate/common/announcement';
+
+describe('Announcement component', () => {
+  const announcementContent = {
+    body: '<p>Bonjour à tous</p>',
+    title: 'Titre de l\'annonce',
```
Comment (amensouissi) [inline assembl/static2/js/app/components/debate/common/announcement.spec.jsx]: title and __typename ==> not used

File: assembl/static2/js/app/components/debate/common/announcement.jsx (hunk 1 above)
Comment (MilanBarande) [reply assembl/static2/js/app/components/debate/common/announcement.jsx]: Ok I figured we have a ```noTitle``` boolean always set to true...

File: assembl/static2/js/app/components/debate/common/announcement.spec.jsx [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+        __typename: 'Idea'
+      };
+      const announcementProps = {
+        announcementContent: announcementContent,
+        idea: threadIdea,
+        isMultiColumns: false
+      };
+      const component = renderer.create(<Announcement {...announcementProps} />);
+      const tree = component.toJSON();
+      expect(tree).toMatchSnapshot();
```
Comment (pascalhuynh) [inline assembl/static2/js/app/components/debate/common/announcement.spec.jsx]: From my point of view, having something matching a snapshot don't really test an expected result.

**Pull request merged**